pkg/gadgettracermanager/containerutils/crio/crio.go

```diff
@@ -75,20 +76,31 @@ func (c *CrioClient) PidFromContainerId(containerID string) (int, error) {
 		Verbose:     true,
 	}
 
-	status, err := c.client.ContainerStatus(context.Background(), request)
+	res, err := c.client.ContainerStatus(context.Background(), request)
 	if err != nil {
 		return -1, err
 	}
 
-	pidStr, ok := status.Info["pid"]
+	info, ok := res.GetInfo()["info"]
 	if !ok {
-		return -1, fmt.Errorf("container status reply from runtime doesn't contain 'pid'")
+		return -1, errors.New("container status reply from runtime doesn't contain extra information")
 	}
 
-	pid, err := strconv.Atoi(pidStr)
+	var infoContent map[string]interface{}
+	err = json.Unmarshal([]byte(info), &infoContent)
 	if err != nil {
-		return -1, err
+		return -1, fmt.Errorf("failed to unmarshal container extra information: %w", err)
+	}
+
+	pidValue, ok := infoContent["pid"]
+	if !ok {
+		return -1, errors.New("container status reply from runtime doesn't contain 'pid' in extra information")
+	}
+
+	pid, ok := pidValue.(float64)
```

Comment:
Why float64? Shouldn't it be `int` (https://github.com/cri-o/cri-o/commit/be8e876cdabec4e055820502fed227aa44971ddc#diff-8b2bcbcf493efc591951918e022b08b127a874c0515351ebcbbeb25a6257b2c8R116)?

Reply:
It is because I am unmarshalling it with `interface{}`, check [here](https://go.dev/play/p/WWJiybLefRv). Instead, if I do it creating an `Info` structure, it would be `int`, check [here](https://go.dev/play/p/nRphSf807iF). For me, it looks like when one uses `interface{}`, every number is considered a `float64`. I noticed the same for other integers (`int64`, `uint32`, etc) inside [`RuntimeSpec`](https://github.com/cri-o/cri-o/blob/be8e876cdabec4e055820502fed227aa44971ddc/server/container_status.go#L117).

I am fine with either way but unless we ask to expose the `Info` structure, we must define it on our own as [here](https://go.dev/play/p/nRphSf807iF) and any change on that structure will break our implementation. Instead, using `interface{}` will not be impacted by changes inside the structure. Notice such structure has already [changed again](https://github.com/cri-o/cri-o/blob/d0df46cfca75c8bc2934ee24c33fbdc933962556/server/container_status.go#L118-L123), the `privileged` field was introduced.

WDYT?

Reply:
> For me, it looks like when one uses `interface{}`, every number is considered a `float64`. 

TIL 

 
> I am fine with either way but unless we ask to expose the `Info` structure, we must define it on our own as [here](https://go.dev/play/p/nRphSf807iF) and any change on that structure will break our implementation. Instead, using `interface{}` will not be impacted by changes inside the structure. Notice such structure has already [changed again](https://github.com/cri-o/cri-o/blob/d0df46cfca75c8bc2934ee24c33fbdc933962556/server/container_status.go#L118-L123), the `privileged` field was introduced.
> 

I think both approaches have the same compatibility problems. They will break if the type of the field we're interested in changes to something that is not compatible (string to int for instance) or the name of the json field changes. Otherwise other changes should not affect it. 

We're only interested in the pid field, something like 

```
type Info struct  {
    Pid int `json:"pid"`
}
```

should work fine. 

IMO it's better to define the structure, so you don't have to worry about interfaces and so on.